module-6/ocp-example.js

```diff
@@ -0,0 +1,157 @@
+// OCP
+// High-level stuff: displaying the UI and markup
+// Low-level stuff: location, settings, ...
+// Protect high-level stuff from changes in low-level stuff
+// If component A should be protected from changes in component B, then component B should depend on component A
+
+const numeralJs = format => number => formatNumber(format, number)
+
+// Responsibility: provide the correct markup for the number
+function NumberFormatter({ format, number }) {
+  return (
+    <div>
+      {numeralJs(format)(number)}
+    </div>
+  )
+}
+
+function SimpleComponent({ number }) {
+  return (
+    <div>
+      The number is: <br />
+      <NumberFormatter format={'0.000'} number={number} />
+    </div>
+  )
+}
+
+// !!! Requirement change: if the country is US we need to insert commas !!!
+// Before: 10000.000
+// After: 10,000.000
+
+// Solution:
+function NumberFormatter({ format, number }) {
+  const adjustedFormat = checkIfUS() ? `0,${format}` : format // 0,0.000
+
+  return (
+    <div>
+      {numeralJs(adjustedFormat)(number)}
+    </div>
+  )
+}
+
+// !!! Requirement change: if the `display negatives in parenthesis` in the settings is set !!!
+// !!! then negatives should be displayed in parenthesis !!!
+// Before: -10000.000
+// After: (10000.000)
```

Comment:
to jest wtf

Reply:
https://www.accountingcoach.com/blog/use-of-parentheses

Reply:
what the actual fuck